### PR TITLE
Crm 16189 5

### PR DESCRIPTION
**Pull request by pradpnayak**

### CODE REVIEW

**Comment by pradpnayak** [comment]
will commit unit test soon

File: tests/phpunit/CRM/Core/OptionGroupTest.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+   * Ensure only one option value exists after calling ensureOptionValueExists.
+   */
+  public function testWeightOptionGroup() {
+    $values = array();
+    $options1 = CRM_Core_OptionGroup::values('activity_type');
+    $options2 = CRM_Core_OptionGroup::values('activity_type', FALSE, FALSE, FALSE, NULL, 'label', TRUE, TRUE, 'value', 'name');
+    // Verify that arrays are equal.
+    $this->assertTrue(($options1 == $options2), "The arrays retrieved should be the same");
+    // Verify sequence is different.
+    $this->assertFalse(($options1 === $options2), "The arrays retrieved should be the same, but in a different order");
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Core/OptionGroupTest.php]: Cool - I just learnt something. I didn't know that == vs === worked that with arrays

**Comment by eileenmcnaughton** [comment]
I just gave this merge on pass because I'm happy the change is safe, makes sense & has adequate (additional) test coverage.

File: CRM/Core/OptionGroup.php [hunk 2]
```diff
@@ -152,7 +153,7 @@ public static function &values(
       $query .= $condition;
     }
 
-    $query .= " ORDER BY v.weight";
+    $query .= " ORDER BY v.{$orderBy}";
```
Comment (JoeMurray) [inline CRM/Core/OptionGroup.php]: Is this just a style improvement? Seems to be functionally equivalent.

**Pull request merged**